internal/resources/metal/device/list_datasource.go

```diff
@@ -11,8 +11,8 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func dataSourceMetalDevices() *schema.Resource {
-	dsmd := dataSourceMetalDevice()
+func ListDataSource() *schema.Resource {
+	dsmd := DataSource()
```

Comment:
I opted to move the `metal_devices` data source into the same directory as the `metal_device` resource and datasource, because I think that's better than putting it in a separate `internal/resources/metal/devices` directory.

_If_ that is agreed to be a good approach, then that raises the question: should `metal_device_bgp_neighbors` _also_ go in this directory, or does it still go in a separate `internal/resources/metal/device_bgp_neighbors` directory?

Reply:
The pluralization of the resource names, as in multi-response datasources, was a subtle point that I missed on first glance of this comment.

AWS followed this pattern. Seems fine to me. 
https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/ec2_instances_data_source.go